crates/ironrdp-web/src/session.rs

```diff
@@ -207,16 +203,12 @@ impl iron_remote_desktop::SessionBuilder for SessionBuilder {
         self.clone()
     }
 
-    fn extension(&self, value: JsValue) -> Self {
-        match serde_wasm_bindgen::from_value::<Extension>(value) {
-            Ok(value) => match value {
-                Extension::KdcProxyUrl(kdc_proxy_url) => self.0.borrow_mut().kdc_proxy_url = Some(kdc_proxy_url),
-                Extension::Pcb(pcb) => self.0.borrow_mut().pcb = Some(pcb),
-                Extension::DisplayControl(use_display_control) => {
-                    self.0.borrow_mut().use_display_control = use_display_control
-                }
-            },
-            Err(error) => error!(%error, "Unsupported extension value"),
+    fn extension(&self, ext: Extension) -> Self {
+        iron_remote_desktop::extension_match! {
+            match ext;
+            |pcb: String| { self.0.borrow_mut().pcb = Some(pcb) };
+            |kdc_proxy_url: String| { self.0.borrow_mut().kdc_proxy_url = Some(kdc_proxy_url) };
+            |display_control: bool| { self.0.borrow_mut().use_display_control = display_control };
         }
```

Comment:
We directly use `JsValue`.
No extra dependency required for that.
For now there is no extension requiring to pass an "object", but if we had one, we could use the [`Reflect` module of `js_sys`](https://docs.rs/js-sys/latest/js_sys/Reflect/index.html) (that we already depend on)